data/document_convention.go

```diff
@@ -0,0 +1,120 @@
+package data
+
+import (
+	"errors"
+	"time"
+)
+
+type BehaviorType uint8
+
+const (
+	LEADER_ONLY BehaviorType = iota
+	LEADER_WITH_FAILOVER
+	LEADER_WITH_FAILOVER_WHEN_REQUEST_TIME_SLA_THRESHOLD_IS_REACHED
+	ROUND_ROBIN
+	ROUND_ROBIN_WITH_FAILOVER_WHEN_REQUEST_TIME_SLA_THRESHOLD_IS_REACHED
+	FASTEST_NODE
+)
+
+var ReadBehaviours = [...]string{
+	"LeaderOnly",
+	"LeaderWithFailover",
+	"LeaderWithFailoverWhenRequestTimeSlaThresholdIsReached",
+	"RoundRobin",
+	"RoundRobinWithFailoverWhenRequestTimeSlaThresholdIsReached",
+	"FastestNode",
+}
+
+var WriteBehaviours = [...]string{
+	"LeaderOnly",
+	"LeaderWithFailover",
+}
+
+type Behaviourer interface {
+	getBehaviourName() string
+}
+
+type Behaviour struct {
+	allowedBehaviours []string
+	behaviorType BehaviorType
+}
+
+type ReadBehaviour struct {
+	behaviour Behaviour
+}
+
+type WriteBehaviour struct {
+	behaviour Behaviour
+}
+
+type DocumentConvention struct {
+	MaxNumberOfRequestsPerSession, MaxIdsToCatch,
+	Timeout, MaxLengthOfQueryUsingGetUrl uint
+	DefaultUseOptimisticConcurrency bool
+	systemDB, IdentityPartsSeparator string
```

Comment:
Yeah, it is not used anywhere despite its presence in python client. I took it without paying attention to it. Will remove.